Sources/BaggageContext/Context.swift

```diff
@@ -0,0 +1,287 @@
+//===----------------------------------------------------------------------===//
+//
+// This source file is part of the Swift Baggage Context open source project
+//
+// Copyright (c) 2020 Moritz Lang and the Swift Baggage Context project authors
+// Licensed under Apache License v2.0
+//
+// See LICENSE.txt for license information
+//
+// SPDX-License-Identifier: Apache-2.0
+//
+//===----------------------------------------------------------------------===//
+
+import Baggage
+import Logging
+
+/// A default `Context` type.
+///
+/// It is a carrier of contextual `Baggage` and related `Logger`, allowing to log and trace throughout a system.
+///
+/// Any values set on the `baggage` will be made accessible to the logger as call-site metadata, allowing it to log those.
+///
+/// ### Logged Metadata and Baggage Items
+///
+/// Please refer to your configured log handler documentation about how to configure which metadata values should be logged
+/// and which not, as each log handler may handle and configure those differently. The default implementations log *all*
+/// metadata/baggage values present, which often is the right thing, however in larger systems one may want to choose a
+/// log handler which allows for configuring these details.
+///
+/// ### Accepting context types in APIs
+///
+/// It is preferred to accept values of `ContextProtocol` in library APIs, as this yields a more flexible API shape,
+/// to which other libraries/frameworks may pass their specific context objects.
+///
+/// - SeeAlso: `Baggage` from the Baggage module.
+/// - SeeAlso: `Logger` from the SwiftLog package.
+public struct Context: ContextProtocol {
+    /// The `Baggage` carried with this context.
+    /// It's values will automatically be made available to the `logger` as metadata when logging.
+    ///
+    /// Baggage values are different from plain logging metadata in that they are intended to be
+    /// carried across process and node boundaries (serialized and deserialized) and are made
+    /// available to instruments using `swift-distributed-tracing`.
+    public var baggage: Baggage
+
+    public var logger: Logger {
+        get {
+            return self._logger.with(self.baggage)
+        }
+        set {
+            self._logger = newValue
+        }
+    }
+
+    private var _logger: Logger
+
+    public init(baggage: Baggage, logger underlying: Logger) {
+        self.baggage = baggage
+        self._logger = underlying
+    }
+
+    public var asBaggageContext: Context {
```

Comment:
`asContext`? also is this required?

Reply:
This exists somewhat defensively, if we need to back into a default context. But in a call today I realized this is cartering to other bad APIs, and we should not make their lifes easier :P 

Removing this thanks